paasta_tools/generate_deployments_for_service.py

```diff
@@ -274,40 +280,22 @@ def get_desired_state(branch: str, remote_refs, deploy_group: str) -> Tuple[str,
         return ('start', None)
 
 
-def get_deployments_dict_from_deploy_group_mappings(deploy_group_mappings, v2_deploy_group_mappings):
+def get_deployments_dict_from_deploy_group_mappings(
+    deploy_group_mappings: Dict[str, V1_Mapping],
+    v2_deploy_group_mappings: V2_Mappings,
+) -> DeploymentsDict:
     return {'v1': deploy_group_mappings, 'v2': v2_deploy_group_mappings}
 
 
-def get_deploy_group_mappings_from_deployments_dict(deployments_dict):
-    try:
-        print("Got v1")
-        return deployments_dict['v1']
-    except KeyError:
-        deploy_group_mappings = {}
-        for deploy_group, image in deployments_dict.items():
-            if isinstance(image, str):
-                deploy_group_mappings[deploy_group] = {
-                    'docker_image': image,
-                    'desired_state': 'start',
-                    'force_bounce': None,
-                }
-        print("Got v2")
-        print(deploy_group_mappings)
-        return deploy_group_mappings
-
-
 def generate_deployments_for_service(service: str, soa_dir: str) -> None:
     try:
         with open(os.path.join(soa_dir, service, TARGET_FILE), 'r') as oldf:
             old_deployments_dict = json.load(oldf)
-            old_mappings = get_deploy_group_mappings_from_deployments_dict(old_deployments_dict)
     except (IOError, ValueError) as e:
-        old_mappings = {}
         old_deployments_dict = {}
     mappings, v2_mappings = get_deploy_group_mappings(
         soa_dir=soa_dir,
         service=service,
-        old_mappings=old_mappings,
```

Comment:
With no mappings, we will be touching this file all the time, right?

Reply:
This variable is entirely unused in `get_deploy_group_mappings`.

Reply:
We have a check immediately after to see whether the old file is the same as the new file:

```
    if deployments_dict != old_deployments_dict:
```